pytorch_lightning/metrics/classification/accuracy.py

```diff
@@ -16,33 +16,52 @@
+from pytorch_lightning.metrics.functional.accuracy import _accuracy_update, _accuracy_compute
 
 
 class Accuracy(Metric):
     r"""
     Computes `Accuracy <https://en.wikipedia.org/wiki/Accuracy_and_precision>`_:
 
-    .. math:: \text{Accuracy} = \frac{1}{N}\sum_i^N 1(y_i = \hat{y_i})
+    .. math::
+        \text{Accuracy} = \frac{1}{N}\sum_i^N 1(y_i = \hat{y}_i)
 
     Where :math:`y` is a tensor of target values, and :math:`\hat{y}` is a
-    tensor of predictions.  Works with binary, multiclass, and multilabel
-    data.  Accepts logits from a model output or integer class values in
-    prediction.  Works with multi-dimensional preds and target.
+    tensor of predictions.
 
-    Forward accepts
+    For multi-class and multi-dimensional multi-class data with probability predictions, the
+    parameter ``top_k`` generalizes this metric to a Top-K accuracy metric: for each sample the
+    top-K highest probability items are considered to find the correct label.
 
-    - ``preds`` (float or long tensor): ``(N, ...)`` or ``(N, C, ...)`` where C is the number of classes
-    - ``target`` (long tensor): ``(N, ...)``
+    This metric generalizes to subset accuracy for multilabel data: for the sample to be counted as
+    correct, all labels in that sample have to be correctly predicted. Consider using :class:`~pytorch_lightning.metrics.classification.HammingLoss`
+    is this is not what you want. In a multi-dimensional multi-class case, the `mdmc_accuracy` parameters
+    gives you a choice between computing the subset accuracy or counting each sample on the extra
+    axis separately.
 
-    If preds and target are the same shape and preds is a float tensor, we use the ``self.threshold`` argument.
-    This is the case for binary and multi-label logits.
-
-    If preds has an extra dimension as in the case of multi-class scores we perform an argmax on ``dim=1``.
+    Accepts all input types listed in :ref:`metrics:Input types`.
 
     Args:
         threshold:
-            Threshold value for binary or multi-label logits. default: 0.5
+            Threshold probability value for transforming probability predictions to binary
+            `(0,1)` predictions, in the case of binary or multi-label inputs. Default: `0.5`
+        top_k:
+            Number of highest probability predictions considered to find the correct label, relevant
+            only for (multi-dimensional) multi-class inputs with probability predictions. The
+            default value (``None``) will be interpreted as 1 for these inputs.
+
+            Should be left at default (``None``) for all other types of inputs.
+        mdmc_accuracy:
+            Determines how should the extra dimension be handled in case of multi-dimensional multi-class
+            inputs. Options are ``"global"`` or ``"subset"``.
+
+            If ``"global"``, then the inputs are treated as if the sample (``N``) and the extra dimension
+            were unrolled into a new sample dimension.
```

Comment:
Can you explicit incorporate that `global` = internally flattens arrays

Reply:
What do you mean by this?

Reply:
If `mdmc_accuracy`=`global` give the same result as calling the metric on `accuracy(preds.flatten(), target.flatten())`

Reply:
Well yes - if your inputs are multi-dimensional multi-class with labels. But this is not true in general - for multi-label inputs this parameter does nothing (subset accuracy is always computed for them), and of course for multi-dimensional multi-class inputs with probability predictions this can't be the case as inputs need to be first transformed into the binarized form.

Reply:
But yeah, I can add this to the description, it should make it clearer

Reply:
Only a minor comment, It is only because this was the behavior of the old accuracy implementation

Reply:
Alright, I updated the description now